src/NUnitFramework/framework/Internal/PropertyBag.cs

```diff
@@ -144,13 +144,16 @@ public TNode AddToXml(TNode parentNode, bool recursive)
             // enumerating dictionary directly with struct enumerator which is fastest
             foreach (var pair in inner)
             {
-                foreach (var value in pair.Value)
+                // Use for-loop to avoid allocating the enumerator
+                var list = pair.Value;
+                var propertyCount = list.Count;
+                for (var i = 0; i < propertyCount; i++)
                 {
                     TNode prop = properties.AddElement("property");
 
                     // TODO: Format as string
                     prop.AddAttribute("name", pair.Key);
-                    prop.AddAttribute("value", value.ToString()!);
+                    prop.AddAttribute("value", list[i]!.ToString()!);
```

Comment:
```suggestion
                    prop.AddAttribute("value", list[i].ToString()!);
```

list elements are not-null. Null suppression operator not needed.